### PR TITLE
refactoring

### PR DESCRIPTION
**Pull request by pldespaigne**
refactoring `file-explorer`

### CODE REVIEW

File: libs/media/src/lib/components/file-explorer/file-explorer.component.html [hunk 1]
```diff
@@ -1,50 +1,55 @@
...
-            </ng-template>
+    </aside>
+
+    <section class="surface">
+
+      <header fxLayout fxLayoutAlign="space-between center">
+
+        <div fxLayout="row wrap" fxLayoutAlign="start center">
+          <ng-container *ngFor="let crumb of breadCrumbs; let isLast = last">
+            <button mat-button color="primary" (click)="goTo(crumb.path)">{{ crumb.name }}</button>
```
Comment (RemcoSimonides) [inline libs/media/src/lib/components/file-explorer/file-explorer.component.html:22]: No issues when trying to navigate to the last crumb in the path?

File: libs/media/src/lib/components/file-explorer/file-explorer.component.html [hunk 2]
```diff
@@ -1,50 +1,55 @@
...
-          </ng-template>
-        </bf-table-filter>
+        <mat-selection-list [multiple]="false">
+          <mat-list-option *ngFor="let directory of activeDirectory.directories" (click)="goTo(directory.path)" >
+            <mat-icon *ngIf="directory.type === 'directory' || directory.multiple" mat-list-icon svgIcon="folder"></mat-icon>
+            <mat-icon *ngIf="directory.type === 'file' && !directory.multiple" mat-list-icon svgIcon="document"></mat-icon>
+            <mat-icon *ngIf="directory.type === 'image' && !directory.multiple" mat-list-icon svgIcon="picture"></mat-icon>
+            <span mat-line>{{ directory.name }}</span>
+          </mat-list-option>
+        </mat-selection-list>
```
Comment (RemcoSimonides) [inline libs/media/src/lib/components/file-explorer/file-explorer.component.html:46]: Thank god for replacing this :) I was also thinking of not using the bf-table-filter anymore

File: libs/media/src/lib/components/file-explorer/file-explorer.component.ts [hunk 3]
```diff
@@ -32,288 +32,168 @@ import {
...
+  ngOnDestroy() {
+    this.dialogSubscription.unsubscribe();
+  }
+
+
+  public goTo(path: number[]) {
+
+    const pathCopy = [...path]; // without a copy we would be modifying the path of the parent directory
+
+    let currentDirectory: Directory | SubDirectoryImage | SubDirectoryFile = this.directories[pathCopy.shift()];
```
Comment (RemcoSimonides) [inline libs/media/src/lib/components/file-explorer/file-explorer.component.ts:94]: I know I did this but maybe we can rename 'SubDirectoryImage' and 'SubDirectoryFile' to 'ImageDirectory' && 'FileDirectory'. 'Sub' is old

File: libs/media/src/lib/components/file-explorer/file-explorer.component.ts [hunk 4]
```diff
@@ -32,288 +32,168 @@ import {
...
+  }
+
+  ngOnDestroy() {
+    this.dialogSubscription.unsubscribe();
+  }
+
+
+  public goTo(path: number[]) {
+
+    const pathCopy = [...path]; // without a copy we would be modifying the path of the parent directory
```
Comment (RemcoSimonides) [inline libs/media/src/lib/components/file-explorer/file-explorer.component.ts:92]: A lot more neat than `Object.assign([], path)` ^^

File: libs/media/src/lib/components/file-explorer/file-explorer.component.ts [hunk 5]
```diff
@@ -32,288 +32,168 @@ import {
...
+        data: {
+          form: mediaForm,
+          ratio: this.activeDirectory.ratio,
+          storagePath: this.activeDirectory.storagePath,
+        },
+      });
     }
 
-    dialog.afterClosed().subscribe(async result => {
+    // on dialog close update teh corresponding document & upload the file if needed
```
Comment (RemcoSimonides) [inline libs/media/src/lib/components/file-explorer/file-explorer.component.ts:179]: teh => the

File: libs/media/src/lib/components/file-explorer/file-explorer.model.ts [hunk 6]
```diff
@@ -8,32 +8,44 @@ import { MovieForm, MovieNotesForm } from "@blockframes/movie/form/movie.form";
...
+interface FileDirectoryBase extends DirectoryBase {
+  /** Wether or not this directory can contain several files *(like stills photo)* or only one *(like movie poster)* */
   multiple: boolean;
   docNameField: string;
   fileRefField: string;
+  /**
+   * Corresponding path to the firebase storage
+   * @note it **SHOULD NOT** start with a `/`
+   * @note it **SHOULD NOT** end with a `/`
+   * @note it **SHOULD NOT** contain the privacy (`public`/`protected`)
```
Comment (RemcoSimonides) [inline libs/media/src/lib/components/file-explorer/file-explorer.model.ts:36]: But we could add it so we can removed `privacy` from this interface?

**Review by RemcoSimonides — CHANGES_REQUESTED** [review]

**Review by RemcoSimonides — APPROVED** [review]

File: libs/media/src/lib/components/file-explorer/file-explorer.component.html (hunk 1 above)
Comment (pldespaigne) [reply libs/media/src/lib/components/file-explorer/file-explorer.component.html:22]: Nope it simply navigate to the same place witch is not even visible

**Pull request merged**